### PR TITLE
Restructure supply mule

### PR DESCRIPTION
**Pull request by Galaxy-Spam**
Inital work for #57

### CODE REVIEW

File: aiscripts/mule.lib.evaluate_tradeoffers.xml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+						<continue />
+					</do_if>
+					<!-- only do any work if the offers are for the same ware -->
+					<do_if value="$someNeed.ware.id != $someSupply.ware.id">
+						<continue />
+					</do_if>
+
+					<!-- how much does the need want? -->
+					<get_ware_reservation object="$someNeed.owner" type="buy" ware="$someNeed.ware" result="$needReservations" />
+					<set_value name="$targetOfferedAmount" exact="$someNeed.desiredamount" />
```
Comment (bhayden53) [inline aiscripts/mule.lib.evaluate_tradeoffers.xml:97]: In the original script there was a different value here depending on whether the buyer was a player-owned object or AI owned. I used $someNeed.amount in the AI case. It's not clear whether this distinction is still needed with clamping now in place.

Comment (Galaxy-Spam) [reply aiscripts/mule.lib.evaluate_tradeoffers.xml:97]: Form my reading of the docs and the experience with clamping trade, clamping trade should be enough here. 
But we should confirm this.

File: aiscripts/mule.lib.debug.dump_tradeoffers.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  <attention min="unknown">
+    <actions>
+			<do_if value="$debugchance">
+				<!-- print headerline & count -->
+				<debug_to_file name="$debugFileName" directory="$debugDirName" text="'----- %s (%s):'.[$headerline, $offers.count]" />
+				<!-- loop over offers and print every single one-->
+				<do_for_each name="$offer" in="$offers">
+					<do_if value="$offer.owner == $offer.seller">
+						<!-- sell offers -->
+						<debug_to_file name="$debugFileName" directory="$debugDirName" text="'%1 wants to sell %2 %3 to %4 for %5 (%6)'.[$offer.seller.knownname, $offer.amount, $offer.ware, $offer.buyer, $offer.unitprice, $offer.relativeprice]" />
```
Comment (bhayden53) [inline aiscripts/mule.lib.debug.dump_tradeoffers.xml]: in this line and line 33 offer.buyer either returns a memory reference or null, what did you intend for that?

Comment (Galaxy-Spam) [reply aiscripts/mule.lib.debug.dump_tradeoffers.xml]: the memory reference can be converted to a name, but for null, that's probably related to the search parameters. I will have a look into it

**Pull request merged**